pyuvdata/tests/__init__.py

```diff
@@ -5,19 +5,245 @@
+        warnings.filterwarnings("ignore", message="numpy.ufunc size changed")
+
+        # Filter iers warnings if iers.conf.auto_max_age is set to None, as we
+        # do in testing if the iers url is down. See conftest.py for more info.
+        if iers.conf.auto_max_age is None:
+            warnings.filterwarnings("ignore", message="failed to download")
+            warnings.filterwarnings("ignore", message="time is out of IERS range")
+
+            test_message = []
+            for message in self.match:
+                if message is None:
+                    test_message.append(False)
+                else:
+                    test_message.append(message.startswith("LST values stored in "))
+            if not any(test_message):
+                warnings.filterwarnings("ignore", message="LST values stored in ")
+
+        return self
+
+    def __exit__(
+        self, exc_type=None, exc_val=None, exc_tb=None,
+    ):
+        if not self._entered:
+            __tracebackhide__ = True
+            raise RuntimeError("Cannot exit %r without entering first" % self)
+
+        super().__exit__(exc_type, exc_val, exc_tb)
+
+        # Built-in catch_warnings does not reset entered state so we do it
+        # manually here for this context manager to become reusable.
+        self._entered = False
+
+        # only check if we're not currently handling an exception
+        if exc_type is None and exc_val is None and exc_tb is None:
+            if self.expected_warning is not None:
+                assert len(self) == len(self.expected_warning), (
+                    f"{len(self.expected_warning)} warnings expected, "
+                    f"{len(self)} warnings issued. The list of emitted warnings is: "
+                    f"{[each.message for each in self]}."
+                )
+
+                for warn_i, exp_warn in enumerate(self.expected_warning):
+                    if not any(issubclass(r.category, exp_warn) for r in self):
+                        __tracebackhide__ = True
+                        raise AssertionError(
+                            "DID NOT WARN. No warnings of type {} was emitted. "
+                            "The list of emitted warnings is: {}.".format(
+                                self.expected_warning, [each.message for each in self]
+                            )
+                        )
+                    elif self.match is not None:
+                        for record in self:
+                            if str(record.message).startswith(
+                                self.match[warn_i]
+                            ) or re.compile(self.match[warn_i]).search(
+                                str(record.message)
+                            ):
+                                if issubclass(
+                                    record.category, self.expected_warning[warn_i]
+                                ):
+                                    break
+                        else:
```

Comment:
I've been looking for a good excuse to use `for/else` statements :tada:

Reply:
Ha! I kind of hate them because I always have to think hard about what they mean, but they are useful sometimes.